Source/automap.cpp

```diff
@@ -831,7 +1304,226 @@ void InitAutomap()
 	size_t tileCount = 0;
 	std::unique_ptr<AutomapTile[]> tileTypes = LoadAutomapData(tileCount);
 	for (unsigned i = 0; i < tileCount; i++) {
-		AutomapTypeTiles[i + 1] = tileTypes[i];
+		AutomapTile tempTileType = tileTypes[i];
+
+		if (IsAnyOf(leveltype, DTYPE_TOWN, DTYPE_CAVES, DTYPE_NEST)) {
+			switch (i + 1) {
+			case 5:

```

Comment:
Rather than doing a switch in a loop, how about simply doing something like this outside the loop:

```c++
if (IsAnyOf(leveltype, DTYPE_TOWN, DTYPE_CAVES, DTYPE_NEST)) {
	tileTypes[4] = { AutomapTile::Types::CaveBottomCorner };
	tileTypes[12] = { AutomapTile::Types::CaveRightCorner };
	tileTypes[13] = { AutomapTile::Types::CaveLeftCorner };
}
```

This is a lot easier to read and is also faster.